src/Utils/FsUtils.php

```diff
@@ -60,4 +60,23 @@ public static function prepareBackupDir($subdir = null)
         return $backup_dir;
     }
 
-}
\ No newline at end of file
+    /**
+     * Returns canonicalized absolute pathname.
+     *
+     * The difference between this and PHP's realpath() is that this will
+     * return the original path even if it doesn't exist.
+     *
+     * @param string $path
+     *   The path being checked.
+     *
+     * @return string
+     *   The canonicalized absolute pathname.
+     */
+    public static function realpath($path)
+    {
+        if (file_exists($path)) {
```

Comment:
You should probably just call `realpath` and return the result if it is not false, and return `$path` otherwise. `realpath` and `file_exists` may have slightly different behaviors (e.g. if $path points to a symlink that points to a file that does not exist)

Reply:
Good point.